test/test_nn.py

```diff
@@ -10511,6 +10511,15 @@ def test(nonlinearity, *args, **kwargs):
         test('threshold', 3, 2)
         test('threshold', 3, 2, inplace=True)
 
+    def test_pooling_shape(self, device):
+        def check(expected_out_shape, sizes, *args, **kwargs):
+            t = torch.randn(sizes, device=device)
+            self.assertEqual(torch.nn.functional.max_pool3d(t, *args, **kwargs).shape, expected_out_shape)
+
+        check((1, 1, 3, 4), (1, 1, 6, 7), kernel_size=1, stride=2, padding=0, ceil_mode=True)
```

Comment:
Add a comment explaining which test case(s) are entirely outside the bounds and what that implies for the shape, vs. those that start with a valid input, vs. those that start in a padded region.

Should one of these test cases have left padding and right padding?

Reply:
I'll add more test cases